### PR TITLE
Add documentation & make targets for starting a local doc site

### PR DESCRIPTION
**Pull request by tstromberg**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -493,3 +493,24 @@ install-kvm-driver: out/docker-machine-driver-kvm2
 release-kvm-driver: install-kvm-driver checksum
 	gsutil cp $(GOBIN)/docker-machine-driver-kvm2 gs://minikube/drivers/kvm/$(VERSION)/
 	gsutil cp $(GOBIN)/docker-machine-driver-kvm2.sha256 gs://minikube/drivers/kvm/$(VERSION)/
+
+site/themes/docsy/assets/vendor/bootstrap/package.js:
+	git submodule update -f --init --recursive
+
+# hugo for generating site previews
+.ONESHELL:
+out/hugo/hugo:
```
Comment (medyagh) [inline Makefile]: maybe add a check to this, only if hugo is not in the path ?

Comment (tstromberg) [reply Makefile]: I thought about this, but didn't want to get into checking if it was the same version and compiled with the same options. This leaves us room to pin to a specific version later.

**Comment by tstromberg** [comment]
> after running `make site` I have the following files changed:
> 
> ```
> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
>         site/resources/
> ```
> 
> that caused my vscode go slow because of tracking the changes
> 
> add site/resources/ to the .gitignore.

Fixed.

**Comment by medyagh** [comment]
make test ( the boilerplate check)  checks for boilerplate in the out/hugo folder need to make boilerplate script not to check in out folder

```

boilerplate missing: /usr/local/google/home/medya/workspace/minikube/./out/hugo/bench.sh
/usr/local/google/home/medya/workspace/minikube/./out/hugo/pull-docs.sh
/usr/local/google/home/medya/workspace/minikube/./out/hugo/bepdock.sh
```

**Comment by tstromberg** [comment]
> make test ( the boilerplate check) checks for boilerplate in the out/hugo folder need to make boilerplate script not to check in out folder
> 
> ```
> 
> boilerplate missing: /usr/local/google/home/medya/workspace/minikube/./out/hugo/bench.sh
> /usr/local/google/home/medya/workspace/minikube/./out/hugo/pull-docs.sh
> /usr/local/google/home/medya/workspace/minikube/./out/hugo/bepdock.sh
> ```

Fixed.

**Pull request merged**